slowquant/qiskit_interface/interface.py

```diff
@@ -719,60 +814,96 @@ def _one_call_sampler_distributions(
             paulis: (List of) Pauli strings to measure.
             run_paramters: List of parameters of each circuit.
             circuits_in: List of circuits
+            overwrite_shots: overwrite QI shot number
 
         Returns:
             Array of quasi-distributions in order of all circuits results for a given Pauli String first.
             E.g.: [PauliString[0] for Circuit[0], PauliString[0] for Circuit[1], ...]
         """
-        if not isinstance(self._primitive, BaseSampler):
-            raise TypeError(f"Expected primitive to be of type BaseSampler got, {type(self._primitive)}")
+        if self._circuit_multipl > 1:
+            shots: int | None = self.max_shots_per_run
```

Comment:
What is going on here?

Reply:
If we need the "shot cheater", we use the "max_shots_per_run" for each circuit and append the identical circuit "_circuit_multipl" times. Thus, we get max_shots_per_run * _circuit_multipl number of shots for a given circuit and Pauli string.